programs/drift/src/instructions/keeper.rs

```diff
@@ -1227,6 +1237,446 @@ pub fn handle_liquidate_spot<'c: 'info, 'info>(
+                ErrorCode::InvalidLiquidateSpotWithSwap,
+                "the liability_token_account passed to SwapBegin and End must match"
+            )?;
+
+            validate!(
+                ctx.accounts.asset_token_account.key() == ix.accounts[9].pubkey,
+                ErrorCode::InvalidLiquidateSpotWithSwap,
+                "the asset_token_account passed to SwapBegin and End must match"
+            )?;
+
+            validate!(
+                ctx.remaining_accounts.len() == ix.accounts.len() - 13,
+                ErrorCode::InvalidLiquidateSpotWithSwap,
+                "begin and end ix must have the same number of accounts"
+            )?;
+
+            for i in 13..ix.accounts.len() {
+                validate!(
+                    *ctx.remaining_accounts[i - 13].key == ix.accounts[i].pubkey,
+                    ErrorCode::InvalidLiquidateSpotWithSwap,
+                    "begin and end ix must have the same accounts. {}th account mismatch. begin: {}, end: {}",
+                    i,
+                    ctx.remaining_accounts[i - 13].key,
+                    ix.accounts[i].pubkey
+                )?;
+            }
+        } else {
+            if found_end {
+                for meta in ix.accounts.iter() {
+                    validate!(
+                        meta.is_writable == false,
+                        ErrorCode::InvalidLiquidateSpotWithSwap,
+                        "instructions after swap end must not have writable accounts"
+                    )?;
+                }
+            } else {
+                let mut whitelisted_programs = vec![
+                    serum_program::id(),
+                    AssociatedToken::id(),
+                    jupiter_mainnet_3::ID,
+                    jupiter_mainnet_4::ID,
+                    jupiter_mainnet_6::ID,
+                ];
+                validate!(
+                    whitelisted_programs.contains(&ix.program_id),
+                    ErrorCode::InvalidLiquidateSpotWithSwap,
+                    "only allowed to pass in ixs to token, openbook, and Jupiter v3/v4/v6 programs"
+                )?;
+
+                for meta in ix.accounts.iter() {
+                    validate!(
+                        meta.pubkey != crate::id(),
+                        ErrorCode::InvalidLiquidateSpotWithSwap,
+                        "instructions between begin and end must not be drift instructions"
+                    )?;
+                }
+            }
+        }
+
+        index += 1;
+    }
+
+    validate!(
+        found_end,
+        ErrorCode::InvalidLiquidateSpotWithSwap,
+        "found no LiquidateSpotWithSwapEnd instruction in transaction"
+    )?;
+
+    Ok(())
+}
+
+#[access_control(
+    liq_not_paused(&ctx.accounts.state)
+)]
+pub fn handle_liquidate_spot_with_swap_end<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, LiquidateSpotWithSwap<'info>>,
+    asset_market_index: u16,
+    liability_market_index: u16,
```

Comment:
would it be a good idea to add `min_amount_out` so liquidator has some control over worst case